data/altered.go

```diff
@@ -0,0 +1,85 @@
+package data
+
+// AlteredAccount is a structure that holds information about an altered account
+type AlteredAccount struct {
+	IsSender        bool
+	IsESDTOperation bool
+	IsNFTOperation  bool
+	TokenIdentifier string
+	NFTNonceString  string
+}
+
+type AlteredAccounts struct {
+	altered map[string][]*AlteredAccount
+}
+
+func NewAlteredAccounts() *AlteredAccounts {
+	return &AlteredAccounts{
+		altered: make(map[string][]*AlteredAccount),
+	}
+}
+
+func (aa *AlteredAccounts) Add(key string, account *AlteredAccount) {
+	_, ok := aa.altered[key]
+	if !ok {
+		aa.altered[key] = make([]*AlteredAccount, 0)
+		aa.altered[key] = append(aa.altered[key], account)
+
+		return
+	}
+
+	isESDTOrNFT := account.IsESDTOperation || account.IsNFTOperation
+	if !isESDTOrNFT {
+		aa.altered[key][0].IsSender = aa.altered[key][0].IsSender || account.IsSender
+		return
+	}
+
+	wasSender := false
+	for _, elem := range aa.altered[key] {
+		newElemIsESDT := account.IsESDTOperation || account.IsNFTOperation
```

Comment:
`newElementIsTokenOperation` and `oldElementIsTokenOperation`

Reply:
done.